sim/mage/talents.go

```diff
@@ -1,74 +1,140 @@
-				spell.BonusHitRating += bonusHit
+			if spell.SpellSchool.Matches(core.SpellSchoolArcane) && spell.Flags.Matches(SpellFlagMage) {
+				spell.ThreatMultiplier *= threatMultiplier
 			}
 		})
 	}
 
+	// Arcane Focus
 	if mage.Talents.ArcaneFocus > 0 {
 		bonusHit := 2 * float64(mage.Talents.ArcaneFocus) * core.SpellHitRatingPerHitChance
 		mage.OnSpellRegistered(func(spell *core.Spell) {
-			if spell.SpellSchool.Matches(core.SpellSchoolArcane) {
+			if spell.SpellSchool.Matches(core.SpellSchoolArcane) && spell.Flags.Matches(SpellFlagMage) {
 				spell.BonusHitRating += bonusHit
 			}
 		})
 	}
 
-	if mage.Talents.PiercingIce > 0 {
-		mage.PseudoStats.SchoolDamageDealtMultiplier[stats.SchoolIndexFrost] *= 1 + 0.02*float64(mage.Talents.PiercingIce)
+	// Magic Absorption
+	if mage.Talents.MagicAbsorption > 0 {
+		magicAbsorptionBonus := 2 * float64(mage.Talents.MagicAbsorption)
+		mage.AddStat(stats.ArcaneResistance, magicAbsorptionBonus)
+		mage.AddStat(stats.FireResistance, magicAbsorptionBonus)
+		mage.AddStat(stats.FrostResistance, magicAbsorptionBonus)
+		mage.AddStat(stats.NatureResistance, magicAbsorptionBonus)
+		mage.AddStat(stats.ShadowResistance, magicAbsorptionBonus)
 	}
 
-	if mage.Talents.FirePower > 0 {
-		mage.PseudoStats.SchoolDamageDealtMultiplier[stats.SchoolIndexFire] *= 1 + 0.02*float64(mage.Talents.FirePower)
+	// Arcane Meditation
+	mage.PseudoStats.SpiritRegenRateCasting += 0.05 * float64(mage.Talents.ArcaneMeditation)
+
+	if mage.Talents.ArcaneMind > 0 {
+		mage.MultiplyStat(stats.Mana, 1.0+0.02*float64(mage.Talents.ArcaneMind))
 	}
 
+	// Arcane Instability
+	mage.AddStat(stats.SpellCrit, float64(mage.Talents.ArcaneInstability)*1*core.SpellCritRatingPerCritChance)
+	mage.PseudoStats.DamageDealtMultiplier *= 1 + .01*float64(mage.Talents.ArcaneInstability)
+}
+
+func (mage *Mage) applyFireTalents() {
+	mage.applyIgnite()
+	mage.applyImprovedScorch()
+	mage.applyMasterOfElements()
+
+	mage.registerCombustionCD()
+
+	// Burning Soul
+	if mage.Talents.BurningSoul > 0 {
+		threatMultiplier := 1 - .15*float64(mage.Talents.BurningSoul)
+		mage.OnSpellRegistered(func(spell *core.Spell) {
+			if spell.SpellSchool.Matches(core.SpellSchoolFire) && spell.Flags.Matches(SpellFlagMage) {
+				spell.ThreatMultiplier *= threatMultiplier
+			}
+		})
+	}
+
+	// Critical Mass
 	if mage.Talents.CriticalMass > 0 {
 		bonusCrit := 2 * float64(mage.Talents.CriticalMass) * core.SpellCritRatingPerCritChance
 		mage.OnSpellRegistered(func(spell *core.Spell) {
-			if spell.SpellSchool.Matches(core.SpellSchoolFire) {
+			if spell.SpellSchool.Matches(core.SpellSchoolFire) && spell.Flags.Matches(SpellFlagMage) {
 				spell.BonusCritRating += bonusCrit
 			}
 		})
 	}
 
-	if mage.Talents.ArcaneMind > 0 {
-		mage.MultiplyStat(stats.Mana, 1.0+0.02*float64(mage.Talents.ArcaneMind))
+	// Fire Power
+	if mage.Talents.FirePower > 0 {
+		mage.PseudoStats.SchoolDamageDealtMultiplier[stats.SchoolIndexFire] *= 1 + 0.02*float64(mage.Talents.FirePower)
```

Comment:
This is a spell mod, only affects mage spells and is additive with other spell mods of the same type.
See https://www.wowhead.com/classic/spell=11124/fire-power